### ISSUE TITLE
[ERROR] [MatrixHttpClient] (REQ-1) <html>

### ISSUE THREAD

**Issue by RampantPanda** [matrixgpt/matrix-chatgpt-bot#213]
Trying to run this with docker-compose.
My docker-compose.yml is here:https://bin.garudalinux.org/?a34b6788ef9aca53#CBxc8yTUyZc8hp1BqaxQwqyTqC3cLfKs8bk1SVRLxBwp
My .env file is here:https://bin.garudalinux.org/?4cd91773510ba87f#GaJnsTDHUB9TT5XKN66fWQYG8fWjTBphabWFSx3b2SoG

When I run docker-compose up -d, I get this:
`$ docker logs matrix-chatgpt-bot 
yarn run v1.22.19
$ node --enable-source-maps dist/index.js
Wed, 16 Aug 2023 19:51:39 GMT [ERROR] [MatrixHttpClient] (REQ-1) <html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.25.1</center>
</body>
</html>

node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "#<IncomingMessage>".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

Node.js v19.9.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
yarn run v1.22.19
$ node --enable-source-maps dist/index.js
Wed, 16 Aug 2023 19:51:41 GMT [ERROR] [MatrixHttpClient] (REQ-1) <html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.25.1</center>
</body>
</html>

node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

`

**Comment by max298** [comment]
This looks like your `MATRIX_HOMESERVER_URL` is not correct, as nginx (I assume you use NGINX as a reverse proxy for synapse) returns 404, so the matrix-bot-sdk is unable to connect to your homeserver. I think the sdk is searching for `MATRIX_HOMESERVER_URL/_matrix/client/something`.

Could you try to open your MATRIX_HOMESERVER_URL/_matrix in any browser? You should see a response like this:
```json
{"errcode":"M_UNRECOGNIZED","error":"Unrecognized request"}
```

**Comment by RampantPanda** [comment]
The unredacted line is:
`MATRIX_HOMESERVER_URL=https://element.norge.chat`
, and the url opens in the browser.

**Comment by max298** [comment]
Right, because that's the URL from element, the web client. From looking at your URL, what you want to use as the `MATRIX_HOMESERVER_URL` is `https://matrix.norge.chat`.

**Comment by RampantPanda** [comment]
Thanks :) Works great now.
